sycl/test-e2e/DeviceImageBackendContent/test_plan.md

```diff
@@ -0,0 +1,83 @@
+# Test plan for [`sycl_ext_oneapi_device_image_backend_content`][spec-link] extension
+
+## Testing scope
+
+### Device coverage
+
+The unit tests should be launched on every supported device configuration we have.
+
+The end-to-end tests, which consist of checking the interoperability with Level Zero
```

Comment:
Is there going to be test coverage for other backends, like cuda and hip?

Reply:
The unit tests will be run on all configurations but the end to end tests will only run on Level Zero and OpenCL because there does not seem to be a well-defined way to retrieve a kernel from a device image executable for other backends.

Reply:
According to the plan unit tests will only do some basic API checks but not verify that obtained result is correct.
Is it possible to use module management APIs https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MODULE.html (like cuModuleLoad*) to check the result?

Reply:
Yes, that can be used and I should definitely add that to the test plan. For example, the test should use `cuModuleLoadData` to load the binary data into a module and verify that `CUDA_SUCCESS` is returned. We cannot go any further unfortunately as there is no portable way to retrieve our kernel from the module as is the case, for example, with Level Zero(see [here](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_free_function_kernels.asciidoc#level-zero-and-opencl-compatibility)). For HIP, the format of the data is not even defined so I don't think there is much to verify there.